File: phalcon/app/library/Safety.php
<?php
namespace app\library;

use app\Env;
use Phalcon\Crypt;

/**
* 安全验证类
*/
class Safety{

  /* 正则 */
  static function isRight($name='',$val=''){
    $data = [
      'uname'=>'/^[a-zA-Z][a-zA-Z0-9\_\@\-\*\&]{3,15}$/',
      'passwd'=>'/^[a-zA-Z0-9|_|@|-|*|&]{6,16}$/',
      'tel'=>'/^1\d{10}$/',
      'email'=>'/^([a-zA-Z0-9_-])+@([a-zA-Z0-9_-])+(\.[a-zA-Z0-9_-])+/',
      'idcard'=>'/^[1-9]\d{7}((0\d)|(1[0-2]))(([0|1|2]\d)|3[0-1])\d{3}$|^[1-9]\d{5}[1-9]\d{3}((0\d)|(1[0-2]))(([0|1|2]\d)|3[0-1])\d{3}([0-9]|X)$/',
    ];
    return preg_match($data[$name],$val)?true:false;
  }

  /* 加密-字符串 */
	static function key($str){
		return md5($str.Env::$key);
	}
	/* 加密-数组 */
	static function keyArray($param=[]){
		ksort($param);
		reset($param);
		$param['sign'] = Env::$key;
		return md5(http_build_query($param));
	}

  /* 加密-Base64 */
  static function encode($param=[]){
    $text = is_array($param)?json_encode($param):$param;
    try{
      $crypt = new Crypt();
      $token = @$crypt->encryptBase64($text, Env::$key);
      $token = str_replace('+','_',$token);
      return $token;
    }catch (\Exception $e){
      return '';
    }
  }

  /* 解密-Base64 */
  static function decode($token=''){
    $token = str_replace('_','+',$token);
    try{
      $crypt = new Crypt();
      $data = json_decode(@$crypt->decryptBase64($token, Env::$key));
      return $data;
    }catch (\Exception $e){
      return '';
    }
  }

}

File: java/src/main/java/vip/webmis/java/library/Safety.java
package vip.webmis.java.library;

import java.util.HashMap;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import vip.webmis.java.Env;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.TypeReference;
import com.auth0.jwt.JWT;
import com.auth0.jwt.algorithms.Algorithm;
import com.auth0.jwt.interfaces.DecodedJWT;
import com.auth0.jwt.exceptions.JWTCreationException;
import com.auth0.jwt.exceptions.JWTDecodeException;

/**
 * 安全验证类
 */
public class Safety {

  /* 正则 */
  public static boolean isRight(String name, String val){
    HashMap<String,Object> data = new HashMap<String,Object>();
    data.put("uname", "^[a-zA-Z][a-zA-Z0-9\\_\\@\\-\\*\\&]{3,15}$");
    data.put("passwd", "^[a-zA-Z0-9|_|@|-|*|&]{6,16}$");
    data.put("tel", "^1\\d{10}$");
    data.put("email", "^([a-zA-Z0-9_-])+@([a-zA-Z0-9_-])+(\\.[a-zA-Z0-9_-])+");
    data.put("idcard", "^[1-9]\\d{7}((0\\d)|(1[0-2]))(([0|1|2]\\d)|3[0-1])\\d{3}$|^[1-9]\\d{5}[1-9]\\d{3}((0\\d)|(1[0-2]))(([0|1|2]\\d)|3[0-1])\\d{3}([0-9]|X)$");
    Pattern pattern = Pattern.compile((String.valueOf(data.get(name))));
    Matcher matcher = pattern.matcher(val);
    return matcher.matches();
  }

  /* 加密 */
  public static String encode(HashMap<String,Object> data){
    String token = null;
    try {
      Algorithm algorithm = Algorithm.HMAC256(Env.key);
      token = JWT.create().withIssuer(JSON.toJSONString(data)).sign(algorithm);
      return token;
    } catch (JWTCreationException exception){
      return token;
    }
  }

  /* 解密 */
  public static HashMap<String,Object> decode(String token){
    HashMap<String,Object> res = null;
    try {
      DecodedJWT jwt = JWT.decode(token);
      String data = jwt.getIssuer();
      return JSON.parseObject(data,new TypeReference<HashMap<String,Object>>(){});
    } catch (JWTDecodeException exception){
      return res;
    }
  }
  
}
